src/workloads/sharding/MultiShardTransactionsWithManyNamespaces.yml

```diff
@@ -0,0 +1,233 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+
+GlobalDefaults:
+  Database0: &Database0 databaseForMultiShardTestingIndex0
+  Database1: &Database1 databaseForMultiShardTestingIndex1
+
+  ApproxDocumentSize: &ApproxDocumentSize 2000  # = 2kB
+  ApproxDocumentSize50Pct: &ApproxDocumentSize50Pct 1000  # = 1kB
+  DocumentCount: &DocumentCount 100_000
+
+  MaxPhase: &MaxPhase 4
+
+  RandomIdFilter: &RandomIdFilter {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+
+  UpdateOperation: &UpdateOperation
+    OperationName: updateOne
+    OperationCommand:
+      Filter: *RandomIdFilter
+      Update: {$inc: {counter: 1}}
+
+  UpdateOperationDatabase0: &UpdateOperationDatabase0
+    <<: *UpdateOperation
+    Database: *Database0
+
+  UpdateOperationDatabase1: &UpdateOperationDatabase1
+    <<: *UpdateOperation
+    Database: *Database1
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhase
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database0
```

Comment:
Since the goal is to measure the impact of writing the `"affectedNamespaces", should we make the total size of this field larger. Based on [this](https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-Restriction-on-Collection-Names) and [this](https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-Length-of-Database-Names), the max size for the namespace of an unsharded collection is 255 bytes and the max size of the database name is 63 bytes. I know that each transaction is already acting on 25 namespaces, but I think we might want to make increase the length of the namespaces for the test collections. Any thoughts?

Reply:
I can make database name larger, but unfortunatelly, database loader have hard-coded collections names as Collection0, Collection1, ... CollectionN.

Reply:
Made database name 63 chars long.

We can add another something like CollectionNamePrefix to Loader/MonotonicLoader class, so we can do larger collection. However, I feel like this should be a separate ticket.
Maybe instead we can increase the number of collections?

Currently we have 50 collections with CollectionXX (12 bytes) name and 63 bytes database name, so (12 + 63) * 50 = 3750 extra bytes per transaction.

How much extra space do you want to test? 4KB? 16KB?

Reply:
> How much extra space do you want to test? 4KB? 16KB?

I'm not sure.

> Currently we have 50 collections with CollectionXX (12 bytes) name and 63 bytes database name, so (12 + 63) * 50 = 3750 extra bytes per transaction.

This is probably fine. Maybe we should aim for the realistic worst case.

Reply:
A googled to double-check: typical size of a block on a modern SSD and HDD is 4kb, so let's try to hit that. It is still an arbitrary constant, but I will raise the number of collections per database to 30, instead of 25.